### PR TITLE
ARROW-13519: [R] Make doc examples less noisy

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Comment by thisisnic** [comment]
@nealrichardson - please can you take a look at this one and let me know what you think; wasn't sure how exactly to strike the balance between assigning the output of some examples to reduce output vs. doing it on too many where it wouldn't make sense.  Is there a rough heuristic I can use to assess this?

File: r/R/ipc_stream.R [hunk 1]
```diff
@@ -68,8 +68,8 @@ write_ipc_stream <- function(x, sink, ...) {
 #' @return A `raw` vector containing the bytes of the IPC serialized data.
 #' @examplesIf arrow_available()
 #' # The default format is "stream"
-#' write_to_raw(mtcars)
-#' write_to_raw(mtcars, format = "file")
+#' mtcars_raw <- write_to_raw(mtcars)
+#' mtcars_raw_file <- write_to_raw(mtcars, format = "file")
```
Comment (nealrichardson) [inline r/R/ipc_stream.R:72]: TBH I don't think we need both of these (and this isn't really a part of our API we want to advertise, it's to support a quirk of something that sparklyr depended on)

```suggestion
```

**Review by nealrichardson — APPROVED** [review]
One suggestion but LGTM otherwise, thanks

File: r/R/compute.R [hunk 2]
```diff
@@ -89,9 +89,8 @@ call_function <- function(function_name, ..., args = list(...), options = empty_
 #' @param ... Additional parameters passed to `grep()`
 #' @return A character vector of available Arrow C++ function names
 #' @examplesIf arrow_available()
-#' list_compute_functions()
-#' list_compute_functions(pattern = "^UTF8", ignore.case = TRUE)
-#' list_compute_functions(pattern = "^is", invert = TRUE)
+#' available_funcs <- list_compute_functions()
```
Comment (nealrichardson) [inline r/R/compute.R:92]: Fix lint warning

```suggestion
#' available_funcs <- list_compute_functions()
```

**Pull request closed without merging**